src/App.tsx

```diff
@@ -1,60 +1,143 @@
+
+      setTitleInput('');
+      setUserSelect(0);
+    }
+
+    if (titleInput.trim().length === 0) {
+      setIsTitleError(true);
+    }
+
+    if (+userSelect === 0) {
+      setIsUserError(true);
+    }
+  };
+
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
       <form action="/api/users" method="POST">
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          <input
+            type="text"
+            data-cy="titleInput"
+            placeholder="Enter a title"
+            value={titleInput}
+            onChange={(event) => {
+              setTitleInput(event.target.value);
+            }}
+          />
+          {
+            isTitleError
+            && (<span className="error">Please enter a title</span>)
+          }
         </div>
 
         <div className="field">
-          <select data-cy="userSelect">
-            <option value="0" disabled>Choose a user</option>
+          <select
+            data-cy="userSelect"
+            value={userSelect}
+            onChange={(event) => {
+              setUserSelect(+event.target.value);
+            }}
+          >
+            <option
+              value="0"
+              disabled
+            >
+              Choose a user
+            </option>
+            {usersFromServer.map(({ id, name }) => (
+              <option
+                key={id}
+                value={id}
+              >
+                {name}
+              </option>
+            ))}
           </select>
-
-          <span className="error">Please choose a user</span>
+          {isUserError && (<span className="error">Please choose a user</span>)}
         </div>
 
-        <button type="submit" data-cy="submitButton">
+        <button
+          type="button"
+          data-cy="submitButton"
+          onClick={addTodo}
+        >
           Add
         </button>
       </form>
 
       <section className="TodoList">
-        <article data-id="1" className="TodoInfo TodoInfo--completed">
-          <h2 className="TodoInfo__title">
-            delectus aut autem
-          </h2>
-
-          <a className="UserInfo" href="mailto:Sincere@april.biz">
-            Leanne Graham
-          </a>
-        </article>
-
-        <article data-id="15" className="TodoInfo TodoInfo--completed">
-          <h2 className="TodoInfo__title">delectus aut autem</h2>
-
-          <a className="UserInfo" href="mailto:Sincere@april.biz">
-            Leanne Graham
-          </a>
-        </article>
-
-        <article data-id="2" className="TodoInfo">
-          <h2 className="TodoInfo__title">
-            quis ut nam facilis et officia qui
-          </h2>
-
-          <a className="UserInfo" href="mailto:Julianne.OConner@kory.org">
-            Patricia Lebsack
-          </a>
-        </article>
+
+        {visibleTodos.map(({
+          userId, id, completed, title,
+        }) => {
```

Comment:
```suggestion
        {visibleTodos.map(({
          userId,
          id,
          completed,
          title,
        }) => {
```

Reply:
Yes, you are right, I forgot to deploy it again 🥲 now it works